index.js

```diff
@@ -198,9 +198,11 @@ class ServerlessPythonRequirements {
       if (!isFunctionRuntimePython(arguments)) {
         return;
       }
+      const _pyprojectTomlToRequirements = () =>
```

Comment:
This seems like a quite weird pattern to me - could you explain the reasoning behind adding logic like that?

Reply:
Briefly, because I needed to reuse the `pyprojectTomlToRequirements` function and it was originally written to leverage JS `this` binding.

The functions used in this block:
```
      return BbPromise.bind(this)
        .then(pipfileToRequirements)
        .then(_pyprojectTomlToRequirements)
        .then(addVendorHelper)
        .then(installAllRequirements)
        .then(packRequirements)
        .then(setupArtifactPathCapturing);
```

are implemented to use the `this` binding pattern, where dependencies/args are passed using the `this` object.  Elsewhere in the project, functions are implemented using an explicit `pluginInstance` context argument.
My understanding is that the former technique is widely considered a legacy JS pattern, whereas the latter is a more acceptable modern practice.

I opted to convert `pyprojectTomlToRequirements` to use the latter explicit context argument so that I could invoke it from a function that was already being called with such a context (`pluginInstance`) arg.
In order to make the resulting `pyprojectTomlToRequirements` compatible with this `BbPromise` call async chain, I needed to convert the local `this` instance into a function arg, in addition to passing an empty string for the `modulePath` argument (since this call context is for the project root directory).
The lambda function I used achieves this adaption.

Reply:
Thanks for clarification - instead of adding this temp var `_pyprojectTomlToRequirements` let's just add an inline function - I don't think we need this one-of temporary variable